server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -559,6 +639,34 @@ private void addSequencesFromServer(
       });
     }
 
+    private Sequence<T> fetchSequenceFromResultLevelCache(
+        final byte[] cachedResult
+    )
+    {
+      if (strategy == null) {
+        return null;
+      }
+      final Function<Object, T> pullFromCacheFunction = strategy.pullFromCache();
+      final TypeReference<Object> cacheObjectClazz = strategy.getCacheObjectClazz();
+      Sequence<Object> cachedSequence = Sequences.simple(() -> {
+        try {
+          if (cachedResult.length == 0) {
+            return Iterators.emptyIterator();
+          }
+
+          return objectMapper.readValues(
+              objectMapper.getFactory().createParser(cachedResult),
+              cacheObjectClazz
+          );
+        }
+        catch (IOException e) {
+          throw new RuntimeException(e);
```

Comment:
Failing to fetch or parse from cache should NOT cause a complete failure of the query. It should continue on and do what it can to compute things.

Reply:
I'm not completely clear where this will be evaluated, and if the exception is caught and ignored or not.